src/DIRAC/FrameworkSystem/Client/ComponentInstaller.py

```diff
@@ -1103,7 +1103,7 @@ def getInstalledComponents(self):
             body = rFile.read()
 
           for cType in self.componentTypes:
-            if body.find('dirac-%s' % (cType)) != -1:
+            if body.find('dirac_%s' % (cType)) != -1:
```

Comment:
Is this correct? I think the `run` scripts need to be fixed instead.

Reply:
The `run` scripts are fixed already.

Reply:
Are they, shouldn't it be `exec dirac-service Production/ProductionManager ...` as `$DIRAC/DIRAC` doesn't exist?
```bash
[dirac@lbcertifdirac70 ~]$ cat /opt/dirac/runit/Production/ProductionManager/run
#!/bin/bash
  rcfile=/opt/dirac/bashrc
  [ -e $rcfile ] && source $rcfile
  #
  exec 2>&1
  #
  [ "service" = "agent" ] && renice 20 -p $$
  #
  #
  exec python $DIRAC/DIRAC/Core/scripts/dirac_service.py Production/ProductionManager --cfg /opt/dirac/pro/etc/Production_ProductionManager.cfg < /dev/null
```

Reply:
For python2 server installations what's set in these `run` files is, of course, working. At the same time you have a point, but that's not for this specific PR, which is to change (again!) all these files.

Reply:
If this change is needed then I think it needs to be applied to v7r2 which generates `run` scripts with underscores?

As soon as #5000 is merged the scripts will contain dashes again so given most people haven't edited their `run` scripts yet for v7r2 isn't it easier to fix them once and edit v7r2 to make run scripts like:

```bash
exec dirac-service Production/ProductionManager --cfg /opt/dirac/pro/etc/Production_ProductionManager.cfg < /dev/null
```

If you agree I'll make the PR to v7r2.

Reply:
> If this change is needed then I think it needs to be applied to v7r2 which generates `run` scripts with underscores?

In fact, I realize that this change is needed if and only if the `run` scripts have been updated according to https://github.com/DIRACGrid/DIRAC/wiki/DIRAC-v7r2#rename-of-scripts. 
The "problem" is that in there we wrote "This is optional in v7r2 but will be required before moving to v7r3.". Which is of course fair but not consistent. Maybe it would be better to force the above changes in v7r2 altogether. Would you agree? @chaen @atsareg

Reply:
That's fine to make it mandatory in v7r2. We still have the `dirac-service` script though, right ? Otherwise it is crutial to edit the files before the deployment. This must be made crystal clear

Reply:
I propose we change it to:
```diff
- This requires the runit scripts to be modified to run `dirac_service.py` instead of `dirac-service.py`.
+ This requires the runit scripts to be modified to run `dirac-service` instead of `python $DIRAC/DIRAC/Core/scripts/dirac-service.py`.
```

This allows it to still be optional in v7r2 (as the wrappers named `dirac-service.py` to `dirac_service.py` still exist) and avoids:
* everyone needing to change this again when testing a Python 3 server in v7r3+
* needing to change the `body.find('dirac_%s' % (cType))` lines in `ComponentInstaller.py `

Reply:
OK I pushed the requested changes. But if we go this direction, then we should backport my last commit to v7r2, right?

Reply:
https://github.com/DIRACGrid/DIRAC/pull/5098

Reply:
Agreed 👍